dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/tracker/export/trackedentity/TrackedEntityRequestParamsMapper.java

```diff
@@ -153,6 +155,60 @@ public TrackedEntityOperationParams map(
     return builder.build();
   }
 
+  private void validateRequestParams(TrackedEntityRequestParams params) throws BadRequestException {
+
+    String violation = null;
+    if (params.getProgram() != null && params.getTrackedEntityType() != null) {
+      violation = "Program and tracked entity cannot be specified simultaneously";
+    }
+
+    if (params.getProgram() == null) {
+      if (params.getProgramStatus() != null) {
+        violation = "Program must be defined when program status is defined";
+      }
+
+      if (params.getFollowUp() != null) {
+        violation = "Program must be defined when follow up status is defined";
+      }
+
+      if (params.getEnrollmentEnrolledAfter() != null) {
+        violation = "Program must be defined when program enrollment start date is specified";
+      }
+
+      if (params.getEnrollmentEnrolledBefore() != null) {
+        violation = "Program must be defined when program enrollment end date is specified";
+      }
+
+      if (params.getEnrollmentOccurredAfter() != null) {
+        violation = "Program must be defined when program incident start date is specified";
+      }
+
+      if (params.getEnrollmentOccurredBefore() != null) {
+        violation = "Program must be defined when program incident end date is specified";
+      }
+    }
+
+    if (params.getEventStatus() != null
+        && (params.getEventOccurredAfter() == null || params.getEventOccurredBefore() == null)) {
+      violation = "Event start and end date must be specified when event status is specified";
+    }
+
+    if (params.getUpdatedWithin() != null
+        && (params.getUpdatedAfter() != null || params.getUpdatedBefore() != null)) {
+      violation =
+          "Last updated from and/or to and last updated duration cannot be specified simultaneously";
+    }
```

Comment:
Can you review the naming here?
For example, we don't have `incident date` anymore and `last updated` is just called updated now.

Reply:
The parameters used in the payload didn't change so they are not in the release notes.
Here it is just a matter to check how things are called in the if statement and change the error message accordly.
I would even use a notation with back quotes to highlight the parameters that needs to be corrected